### PR TITLE
Don't enforce returns and yields in abstract methods

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

Closes https://github.com/astral-sh/ruff/issues/12685.

### CODE REVIEW

File: crates/ruff_linter/src/rules/pydoclint/rules/check_docstring.rs [hunk 1]
```diff
@@ -702,6 +714,11 @@ pub(crate) fn check_docstring(
         return;
     }
 
+    // Ignore abstract methods.
+    if visibility::is_abstract(&function_def.decorator_list, checker.semantic()) {
+        return;
+    }
```
Comment (charliermarsh) [inline crates/ruff_linter/src/rules/pydoclint/rules/check_docstring.rs:720]: I guess we could enforce _some_ of the rules, e.g., if there's a default implementation that has a raise, and it's not documented? So, only omit the "extraneous" rules?

**Review by MichaReiser — APPROVED** [review]

File: crates/ruff_linter/src/rules/pydoclint/rules/check_docstring.rs (hunk 1 above)
Comment (AlexWaygood) [reply crates/ruff_linter/src/rules/pydoclint/rules/check_docstring.rs:720]: Yeah, I think I'd continue to enforce the rules that say "you returned/raised/yielded something but didn't document it", and I'd skip the ones that say "you documented that it returns/raises/yields something it doesn't". Abstract methods can have implementations that are called using `super()` from concrete methods in subclasses

Comment (charliermarsh) [reply crates/ruff_linter/src/rules/pydoclint/rules/check_docstring.rs:720]: Kk

File: crates/ruff_linter/src/rules/pydoclint/rules/check_docstring.rs [hunk 2]
```diff
@@ -25,6 +25,8 @@ use crate::rules::pydocstyle::settings::Convention;
 /// Docstrings missing return sections are a sign of incomplete documentation
 /// or refactors.
 ///
+/// This rule is not enforced for abstract methods and stubs functions.
```
Comment (AlexWaygood) [inline crates/ruff_linter/src/rules/pydoclint/rules/check_docstring.rs:28]:
```suggestion
/// This rule is not enforced for stub functions.
```

**Pull request merged**